### PR TITLE
{numlib}[foss/2019a,intel/2019a] CGAL v4.14

### PR DESCRIPTION
**Pull request by furstj**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name = 'CGAL'
+version = '4.11.3'
```
Comment (Micket) [inline easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb]: Any reason to stick with the 4.11 release? There are releases up to 4.14, so I think we should go for one of
```
    CGAL 4.13.1 (April 2019)
    CGAL 4.12.2 (April 2019)
    CGAL 4.14 (March 2019)
```
https://www.cgal.org/releases.html

File: easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    ('Boost', '1.70.0'),
+    ('MPFR', '4.0.2'),
+    ('GMP', '6.1.2'),
+    ('libGLU', '9.0.0'),
+    ('Qt5', '5.12.3'),
+]
+
+builddependencies = [
+    ('CMake', '3.13.3'),
+    ('Eigen', '3.3.7'),
```
[the remark this answers is not in the document]
Comment (furstj) [reply easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb]: Yes, but ('Eigen', '3.3.7') requires CMake 3.9.6 which in turn requires old ncurses 5.9. Unfortunatelly ncurses 5.9 didn't compile on my system...

[the remark this answers is not in the document]
Comment (Micket) [reply easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb]: Just locally change that config to use something like 3.12.1. It's just a build-dep. It won't affect anything after Eigen has been installed.

[the remark this answers is not in the document]
Comment (akesandgren) [reply easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb]: It doesn't strictly require that version of CMake, you could just as well take a copy of that eigen-3.3.7.eb easycofig and change the CMake to something newer, like 3.12.1 and build Eigen.

File: easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb (hunk 1 above)
Comment (furstj) [reply easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb]: There was a change in build process between 4.11 and 4.12. I'm afraid that 4.12 would require also a change in easyblock which is too complicated task for me. At least simply upgrading the library version fails during sanity check.

File: easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb (hunk 2 above)
[the remark this answers is not in the document]
Comment (furstj) [reply easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb]: OK, I will do it. Thanks for the hint :).

File: easybuild/easyconfigs/e/Eigen/Eigen-3.3.7-GCCcore-8.2.0.eb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+name = 'Eigen'
+version = '3.3.7'
+
+homepage = 'http://%(namelower)s.tuxfamily.org/index.php?title=Main_Page'
+
+description = """Eigen is a C++ template library for linear algebra: matrices, vectors, numerical solvers,
+ and related algorithms."""
+
+# only includes header files, so no need for a non-dummy toolchain
+toolchain = {'name': 'GCCcore', 'version': '8.2.0'}
```
[the remark this answers is not in the document]
Comment (furstj) [reply easybuild/easyconfigs/e/Eigen/Eigen-3.3.7-GCCcore-8.2.0.eb]: Dropped.

File: easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb]: +1 on makes the necessary changes to the CGAL easyblock to support latest, looks like sticking would result in lagging behind a bit too much...

@furstj Do you have more details on how the sanity check failed exactly?

Comment (akesandgren) [reply easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb]: Looking into this

Comment (akesandgren) [reply easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb]: The fix for this is in the easyblock is already merged, see https://github.com/easybuilders/easybuild-easyblocks/pull/1679

File: easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+checksums = ['3ed80bc87f00eb3c72357f5842907f828f8ab8fc66053351aab76aa10566981a']
+
+dependencies = [
+    ('zlib', '1.2.11'),
+    ('Python', '3.7.2'),
+    ('Boost', '1.70.0'),
+    ('MPFR', '4.0.2'),
+    ('GMP', '6.1.2'),
+    ('libGLU', '9.0.0'),
+    ('Qt5', '5.12.3'),
```
Comment (akesandgren) [inline easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb]: Mesa is a direct dependency so it should be listed.
('Mesa', '19.0.1'),

File: easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+]
+
+builddependencies = [
```
Comment (akesandgren) [inline easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb]: Move builddeps above dependencies

File: easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-intel-2019a-Python-3.7.2.eb [hunk 6]
```diff
@@ -0,0 +1,37 @@
+name = 'CGAL'
+version = '4.11.3'
```
Comment (akesandgren) [inline easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-intel-2019a-Python-3.7.2.eb]: Same here, use 4.14

File: easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-intel-2019a-Python-3.7.2.eb [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+checksums = ['3ed80bc87f00eb3c72357f5842907f828f8ab8fc66053351aab76aa10566981a']
+
+dependencies = [
+    ('zlib', '1.2.11'),
+    ('Python', '3.7.2'),
+    ('Boost', '1.70.0'),
+    ('MPFR', '4.0.2'),
+    ('GMP', '6.1.2'),
+    ('libGLU', '9.0.0'),
+    ('Qt5', '5.12.3'),
```
Comment (akesandgren) [inline easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-intel-2019a-Python-3.7.2.eb]: Add ('Mesa', '19.0.1'),

File: easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-intel-2019a-Python-3.7.2.eb [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+]
+
+builddependencies = [
```
Comment (akesandgren) [inline easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-intel-2019a-Python-3.7.2.eb]: Move a bove dependencies.

File: easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb]: In that case: @furstj please update to latest version of `CGAL`, and use `cgal.py` from https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/c/cgal.py (which you can grab via 'Raw' and supply to `eb` via `--include-easyblocks`)

**Comment by furstj** [comment]
@akesandgren thanks for hint on new cgal.py. It works fine. Your comments are addressed in latest commit.

File: easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb (hunk 1 above)
Comment (furstj) [reply easybuild/easyconfigs/c/CGAL/CGAL-4.11.3-foss-2019a-Python-3.7.2.eb]: @Micket , your comments are resolved in latest commit - CGAL is updated to v4.14 (thanks to akesandgren).

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/f8edb353786658077cca5c94815053d2 for a full test report.

**Comment by akesandgren** [comment]
Going in, thanks @furstj!

**Pull request merged**